UnitTestWorkshopTests/CreateShopTest.swift

```diff
@@ -14,7 +14,199 @@ import RxCocoa
 
 private class CreateShopTest: CreateShopViewModelBaseTest {
     override func spec() {
+        let useCase = CreateShopUsecase()
         
+        beforeEach {
+            self.setupBinding(viewModel: CreateShopViewModel(useCase: useCase))
+        }
+        
+        describe("Register new shop") {
+            
+            // MARK: Typing Shop Name Test Case
+            context("Typing Shop Name that not available") {
+                beforeEach {
+                    useCase.checkShopNameAvailability = { _ in
+                       .just(false)
+                    }
+                }
+                
+                it("Invalid Shopname, still generate the domain name") {
+                    self.shopNameSubject.onNext("supergadgettt")
+                    self.domainNameValue.assertValue("supergadgettt-4")
+                    self.shopNameError.assertValue(ShopError.notAvailable.message)
+                    self.shopNameErrorIsHidden.assertLastValue(false)
+                }
```

Comment:
onNext lebih disaranin untuk ditaruh diluar "it" dik. dalam scope "it" biasanya buat assert aja
dan bisa dibikin verbose per output
misalnya begini
```swift

context("User input shop name that already taken") {
   beforeEach {
      self.shopNameSubject.onNext("supergadgettt")
   }
   it("Should display shop domain") {
      self.domainNameValue.assertValue("supergadgettt-4")
   }
   it("Should display error shop name is not available") {
      self.shopNameError.assertValue(ShopError.notAvailable.message)
   }
   it("Should display shop name error") {
      self.shopNameErrorIsHidden.assertLastValue(false)
   }
}
```

Reply:
okey noted wan, gw coba benerin ya @ihartono